lib/crypt_keeper/lazy_string.rb

```diff
@@ -0,0 +1,59 @@
+module CryptKeeper
+  # LazyString is a wrapper around a String instance to avoid immediatelly
+  # decrypting the String value. Decryption operations can be slow, so by
+  # using the LazyString, decryption is only triggered if the String values
+  # are actually accessed.
+  class LazyString < BasicObject
+    attr_accessor :provider, :encrypted_string
+
+    # Public: Initializes the LazyString.
+    #
+    # provider - the CryptKeeper provider instance
+    # encrypted_string - the encrypted string instance
+    def initialize(provider, encrypted_string)
+      @provider = provider
+      @encrypted_string = encrypted_string
+    end
+
+    # Public: Overrides == to compare the decrypted string to the other object.
+    #
+    # other - the other object to compare this to
+    #
+    # Returns a boolean.
+    def ==(other)
+      decrypted_string == other
+    end
+
+    # Public: Implements to_str for String compatibility.
+    #
+    # Returns the decrypted string.
+    def to_str
+      decrypted_string
+    end
+
+    # Public: Implements to_ary for String compatibility.
+    #
+    # Returns the decrypted string.
```

Comment:
it returns nil, not the decrypted string

Reply:
btw, why to you need to implement this method and not just handle it in the method_missing

Reply:
@rubemz Rspec expects this method for some reason, I added it for now to make it shut up, but I'll investigate more.